clusters/app.ci/quayio-pull-through-cache/configs.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: v1
+kind: List
+items:
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: quayio-pull-through-cache-config
+    namespace: ci
+  data:
+    config.yaml: |
+      version: 0.1
+      log:
+        fields:
+          service: registry
+      storage:
+        cache:
+          blobdescriptor: inmemory
+        filesystem:
+          rootdirectory: /var/lib/registry
+      http:
+        addr: :5000
+        headers:
+          X-Content-Type-Options: [nosniff]
+      health:
+        storagedriver:
+          enabled: true
+          interval: 10s
+          threshold: 3
+      htpasswd:
+        realm: basic-realm
+        path: /etc/htpasswd
+      proxy:
+        remoteurl: https://quay.io
+      s3:
+        encrypt: true
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: quayio-pull-through-cache-exec
+    namespace: ci
+  data:
+    start.sh: |
```

Comment:
I think that is simplier

Reply:
The downside is that if the script is changed, the pod will not automatically restart. The same goes for the username and password. If the secret changes, the pod should exit so that the deployment will relaunch with the new secret value.

Mounting as mentioned here: https://github.com/openshift/release/pull/32402/files#r974317908 and monitoring its md5 could provide a way to trigger that exit.